signer/src/bitcoin/validation.rs

```diff
@@ -0,0 +1,434 @@
+//! validation of bitcoin transactions.
+
+use bitcoin::relative::LockTime;
+use bitcoin::Amount;
+use bitcoin::OutPoint;
+
+use crate::context::Context;
+use crate::error::Error;
+use crate::keys::PublicKey;
+use crate::storage::model::BitcoinBlockHash;
+use crate::storage::model::BitcoinTx;
+use crate::storage::model::QualifiedRequestId;
+use crate::DEPOSIT_LOCKTIME_BLOCK_BUFFER;
+
+/// The necessary information for validating a bitcoin transaction.
+#[derive(Debug, Clone)]
+pub struct BitcoinTxContext {
+    /// This signer's current view of the chain tip of the canonical
+    /// bitcoin blockchain. It is the block hash and height of the block on
+    /// the bitcoin blockchain with the greatest height. On ties, we sort
+    /// by the block hash descending and take the first one.
+    pub chain_tip: BitcoinBlockHash,
+    /// The block height of the bitcoin chain tip identified by the
+    /// `chain_tip` field.
+    pub chain_tip_height: u64,
+    /// How many bitcoin blocks back from the chain tip the signer will
+    /// look for requests.
+    pub tx: BitcoinTx,
+    /// The requests
```

Comment:
```suggestion
    /// The requests associated with the current transaction
```

Reply:
ugh 👍🏽

Reply:
Fixed in https://github.com/stacks-network/sbtc/pull/744/commits/1da998d1d1db6b789b67ee90d1ff08d57a170a0e